massgen/orchestrator.py

```
@@ -1031,6 +1062,240 @@ def _drain_pending_criteria_proposals(self) -> None:
+        if coord is None or getattr(coord, "criteria_mode", "static") != "bootstrap_subagent":
+            return 0
+        if not current_answers:
+            return 0
+        # Signature is content-aware via a short SHA hash of each answer so the
+        # gate fires on every round where any agent's answer changes. (Keying
+        # only on `current_answers.keys()` would fire once per session, since
+        # the agent ID set is stable across rounds.)
+        import hashlib
+
+        content_signature = tuple(
+            sorted((aid, hashlib.sha1(str(content).encode("utf-8")).hexdigest()[:16]) for aid, content in current_answers.items()),
```

Ruff (0.15.12)

[error] Probable use of insecure hash functions in `hashlib`: `sha1`

(S324)